File: my-folder/0409-longest-palindrome/solution.java
import java.util.HashMap;

class Solution {
    public int longestPalindrome(String s) {
        HashMap<Character, Integer> letters = new HashMap<>();
        char[] charArr = s.toCharArray();
        for (Character c : charArr) {
            if (letters.containsKey(c)) {
                letters.replace(c, letters.get(c) + 1);
            } else {
                letters.put(c, 1);
            }
        }
        System.out.println(letters);
        int ans = 0;
        int highestOdd = 0;
        for (Integer value : letters.values()) {
            if (value % 2 == 0) {
                ans += value;
            } else if (value > highestOdd) {
                if (highestOdd != 0) {
                    ans += highestOdd - 1;
                }
                highestOdd = value;
            } else {
                ans += value - 1;
            }
        }
        return ans + highestOdd;
    }
}

File: my-folder/0409-longest-palindrome/solution.py
class Solution(object):
    def longestPalindrome(self, s):
        """
        :type s: str
        :rtype: int
        """
        letters = set()
        length = 0
        for letter in s:
            if letter not in letters:
                letters.add(letter)
            else:
                letters.remove(letter)
                length += 2
        if len(letters) > 0:
            return length + 1
        return length
